src/CommunityToolkit.Maui.Core/Views/SemanticOrderView/MauiSemanticOrderView.ios.cs

```diff
@@ -0,0 +1,55 @@
+﻿using System.Collections.Generic;
+using Foundation;
+using Microsoft.Maui.Platform;
+using UIKit;
+
+namespace CommunityToolkit.Maui.Core.Views;
+
+/// <summary>
+/// The native implementation of the SemanticOrderView control.
+/// </summary>
+public class MauiSemanticOrderView : ContentView, IUIAccessibilityContainer
+{
+	ISemanticOrderView? virtualView;
+	internal ISemanticOrderView? VirtualView
+	{
+		get => virtualView;
+		set
+		{
+			virtualView = value;
+			UpdateViewOrder();
+		}
+	}
+
+	internal void UpdateViewOrder()
+	{
+		var result = GetAccessibilityElements();
+
+		if (result != null)
+		{
+			this.SetAccessibilityElements(NSArray.FromNSObjects(result.ToArray()));
+		}
+	}
+
+	List<NSObject>? GetAccessibilityElements()
```

Comment:
These will be converted to an array, can we change the return type to be an `IEnumerable<NSObject>?` ? That way we don't allocate the List in the memory, just the array that will be consumed. Something like that

```csharp
	internal void UpdateViewOrder()
	{
		var result = GetAccessibilityElements();

		if (result.Any())
		{
			this.SetAccessibilityElements(NSArray.FromNSObjects(result.ToArray()));
		}
	}

	IEnumerable<NSObject> GetAccessibilityElements()
	{
		if (VirtualView is null)
		{
			yield break;
		}

		var viewOrder = VirtualView.ViewOrder;

		foreach (var view in viewOrder)
		{
			if (view.Handler is IPlatformViewHandler pvh &&
				pvh.PlatformView is not null)
			{
				yield return pvh.PlatformView;
			}
		}
	}
```